src/languages/scala.js

```diff
@@ -151,11 +175,23 @@ export default function(hljs) {
     beginScope: { 2: "keyword", }
   };
 
+  const SOFT_MODIFIER = {
+    begin: [
+      /\b(?<!\.)(infix|inline|opaque|open|transparent)\b/,
```

Comment:
We can't use look-behind because it's not supported in Safari, sorry.

Reply:
Is the "followed by" part of the rule on 182 not enough to prevent false positives?

Reply:
A false positive case without look-behind is as below:

```
a.infix == b.infix

opaque type MyInt = Int
```

expected:
```html
a.infix == b.infix

<span class="hljs-keyword">opaque</span> <span class="hljs-class"><span class="hljs-keyword">type</span> <span class="hljs-title">MyInt</span> </span>= <span class="hljs-type">Int</span>
```

actual:
```html
a.infix == <span class="hljs-keyword">b.infix</span>

<span class="hljs-keyword">opaque</span> <span class="hljs-class"><span class="hljs-keyword">type</span> <span class="hljs-title">MyInt</span> </span>= <span class="hljs-type">Int</span>
```


but in the next case, both `transparent` and `trait` should be keywords.
```
transparent
trait Foo
```

Would you give me some advice to handle this case?

Reply:
Removed look-behind.

Reply:
Looking for another corner case...

Reply:
For `.blah` I'd add a "property" rule that would catch those and highlight them as properties so they don't trigger false positives as keywords.

Reply:
Thanks. `a.infix` can also be written as `a infix` in a certain condition, so I'v just revised the matching rule in the lower level.

Reply:
> Thanks. a.infix can also be written as a infix in a certain condition

Well if you didn't want to be inconsistent (between those two cases) you could match on `.thing` but NOT highlight it (just to avoid the false positive).  This is a very common strategy used in many grammars to handle easy false positives.